cocos/core/director.ts

```diff
@@ -727,21 +688,84 @@ export class Director extends EventTarget {
         }
     }
 
-    private _initOnRendererInitialized () {
+    /**
+     * @internal
+     */
+    public init () {
         this._totalFrames = 0;
         this._paused = false;
-
         // Scheduler
         // TODO: have a solid organization of priority and expose to user
         this.registerSystem(Scheduler.ID, this._scheduler, 200);
-
+        this._root = new Root(deviceManager.gfxDevice);
+        const rootInfo = {};
+        this._root.initialize(rootInfo);
+        for (let i = 0; i < this._systems.length; i++) {
+            this._systems[i].init();
+        }
         this.emit(Director.EVENT_INIT);
     }
 
-    private _init () {
-        this._root = new Root(game._gfxDevice!);
-        const rootInfo = {};
-        this._root.initialize(rootInfo);
+    //  @ Persist root node section
+    /**
+     * @en
+     * Add a persistent root node to the game, the persistent node won't be destroyed during scene transition.<br>
+     * The target node must be placed in the root level of hierarchy, otherwise this API won't have any effect.
+     * @zh
+     * 声明常驻根节点，该节点不会在场景切换中被销毁。<br>
+     * 目标节点必须位于为层级的根节点，否则无效。
+     * @param node - The node to be made persistent
+     */
+    public addPersistRootNode (node: Node) {
+        if (!legacyCC.Node.isNode(node) || !node.uuid) {
+            warnID(3800);
+            return;
+        }
+        const id = node.uuid;
+        if (!this._persistRootNodes[id]) {
+            const scene = legacyCC.director._scene;
+            if (legacyCC.isValid(scene)) {
+                if (!node.parent) {
+                    node.parent = scene;
+                    node._originalSceneId = scene.uuid;
+                } else if (!(node.parent instanceof legacyCC.Scene)) {
+                    warnID(3801);
+                    return;
+                } else if (node.parent !== scene) {
+                    warnID(3802);
+                    return;
+                } else {
+                    node._originalSceneId = scene.uuid;
+                }
+            }
+            this._persistRootNodes[id] = node;
+            node._persistNode = true;
+            legacyCC.assetManager._releaseManager._addPersistNodeRef(node);
+        }
+    }
+
+    /**
+     * @en Remove a persistent root node.
+     * @zh 取消常驻根节点。
+     * @param node - The node to be removed from persistent node list
+     */
+    public removePersistRootNode (node: Node) {
+        const id = node.uuid || '';
+        if (node === this._persistRootNodes[id]) {
+            delete this._persistRootNodes[id];
+            node._persistNode = false;
```

Comment:
I think, instead of visitiing private fields of `Node` twices or more:

```
class Node {
  /** @internal Mark this node as persisting node */
  public void _persist(originalScene: string) {
    this._originalSceneId = originalScene;
    this._persistNode = true;
  }

  /** @internal Unmark this node as persisting node */
  public _desist () {
    this._originalSceneId = '';
    this._persistNode = false;
  }

  private _originalSceneId = '';
  private _persistNode = false;
}
```

Reply:
I think this is OK, but I don't want to introduce additional module adjustments in this pr, I want to wait it until the Node adjustment cocos/3d-tasks#9283